hack/install-vm.sh

```diff
@@ -141,17 +150,246 @@ EOF
     systemctl status "$ANTREA_AGENT"
 }
 
+# Generic function to execute command with timeout
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date), exiting"
+        exit 2
+    fi
+}
+
+# Install docker as container runtime
+install_container_runtime() {
+    echo "Installing container runtime on $OS_NAME"
+    case "$OS_NAME" in
+        ubuntu)
+        # Uninstall older versions of Docker(called docker, docker.io, or docker-engine)
+        docker_prereq_packages="docker docker-engine docker.io containerd runc"
+        docker_install_packages="ca-certificates curl gnupg lsb-release"
+        docker_gpg_key="/usr/share/keyrings/docker-archive-keyring.gpg"
+        docker_download_url="https://download.docker.com/linux/ubuntu"
+        cmd="apt-get remove ${docker_prereq_packages}"
+        execute_command "$cmd" 1m
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        cmd="apt-get install -y ${docker_install_packages}"
+        execute_command "$cmd" 1m
+        rm -rf ${docker_gpg_key}
+        cmd="curl -fsSL ${docker_download_url}/gpg | gpg --dearmor -o ${docker_gpg_key}"
+        execute_command "$cmd" 1m
+        # Setup a stable repository
+        echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=${docker_gpg_key}] ${docker_download_url} \
+        $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        cmd="apt-get install -y docker-ce docker-ce-cli containerd.io"
+        execute_command "$cmd" 2m
+        ;;
+    esac
+}
+
+# Install all pre-requisite packages needed for install
+install_docker_prerequisite_packages() {
+    set +eo pipefail
+    # Install docker, docker-compose only when running services in containers
+    if [ "$DOCKERIZE" = true ]; then
```

Comment:
Isnt this function called for dockerize case. why have check again in this function